### PR TITLE
Additional ESLint rules

### PR DESCRIPTION
**Pull request by lemonmade**
I had a bunch of ideas yesterday so I just built them all. Just get to this the you can :) @GoodForOneFare

### CODE REVIEW

File: packages/eslint-plugin-shopify/docs/rules/class-property-semi.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  bar = true;
+  static baz = false;
+}
+```
+
+The success/ failure cases are reversed when the argument is explicitly `'never'`.
+
+## When Not To Use It
+
+If you do not wish to enforce semicolons for class proeprties, then you can safely disable this rule.
```
Comment (GoodForOneFare) [inline packages/eslint-plugin-shopify/docs/rules/class-property-semi.md]: Typo: `proeprties `

File: packages/eslint-plugin-shopify/docs/rules/restrict-full-import.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Prevents importing the entirety of a package. (restrict-full-import)
+
+Importing the entirety of a large module can be undesirable because it becomes harder to track what properties are being used.
+
+## Rule Details
+
+This rule aims to restrict the imports of specified modules to only be an import of specific properties. This does **not** mean that you are not importing from the module, it simply means that you only extract the properties you need.
```
Comment (GoodForOneFare) [inline packages/eslint-plugin-shopify/docs/rules/restrict-full-import.md]: The first sentence isn't easy to read.  Why is it just aiming?  I'd prefer something like, "This rule prevents default imports for a configurable list of modules."

Second sentence's "does not mean that you are not" would be easier to parse without the double negative, e.g.,: "This is **not** a module blacklist; individual properties can still be imported."

File: packages/eslint-plugin-shopify/docs/rules/restrict-full-import.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+var _ = require('something-else');
+var chain = require('lodash').chain;
+var chain = require('lodash/chain');
+var {chain} = require('lodash');
+```
+
+## When Not To Use It
+
+If you do not want to restrict from full imports from modules, you can safely disable this rule.
```
Comment (GoodForOneFare) [inline packages/eslint-plugin-shopify/docs/rules/restrict-full-import.md]: Reword: `restrict from full`.

File: packages/eslint-plugin-shopify/lib/rules/class-property-semi.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        fix: function(fixer) {
+          fixer.insertTextAfter(lastToken, ';');
+        },
+      });
+    } else if (!always && hasSemicolon) {
+      context.report({
+        node: node,
+        message: 'Extra semicolon.',
+        fix: function(fixer) {
+          fixer.remove(lastToken);
```
Comment (GoodForOneFare) [inline packages/eslint-plugin-shopify/lib/rules/class-property-semi.js]: Whoa! Nice!

**Comment by GoodForOneFare** [comment]
Quibbles over wording, but the code is solid 👍

File: packages/eslint-plugin-shopify/docs/rules/restrict-full-import.md (hunk 2 above)
Comment (lemonmade) [inline packages/eslint-plugin-shopify/docs/rules/restrict-full-import.md]: Good call 👍

**Pull request merged**